test/integration/sessions/sessions.test.ts

```diff
@@ -72,93 +73,108 @@ describe('Sessions Spec', function () {
       });
     });
 
-    describe('withSession', {
-      metadata: {
-        requires: {
-          mongodb: '>=3.6.0'
-        }
-      },
-      test: function () {
-        beforeEach(function () {
-          return test.setup(this.configuration);
-        });
+    describe('withSession', function () {
+      let client: MongoClient;
+      beforeEach(async function () {
+        client = await this.configuration.newClient().connect();
+      });
+
+      afterEach(async function () {
+        await client?.close();
+      });
 
-        [
-          {
-            description: 'should support operations that return promises',
-            operation: client => session => {
-              return client.db('test').collection('foo').find({}, { session }).toArray();
-            }
-          },
-          // {
-          //   nodeVersion: '>=8.x',
-          //   description: 'should support async operations',
-          //   operation: client => session =>
-          //     async function() {
-          //       await client
-          //         .db('test')
-          //         .collection('foo')
-          //         .find({}, { session })
-          //         .toArray();
-          //     }
-          // },
-          {
-            description: 'should support operations that return rejected promises',
-            operation: (/* client */) => (/* session */) => {
-              return Promise.reject(new Error('something awful'));
-            }
-          },
-          {
-            description: "should support operations that don't return promises",
-            operation: (/* client */) => (/* session */) => {
-              setTimeout(() => {});
-            }
-          },
-          {
-            description: 'should support operations that throw exceptions',
-            operation: (/* client */) => (/* session */) => {
-              throw new Error('something went wrong!');
-            }
+      const tests = [
+        {
+          description: 'should resolve non-async callbacks that return promises',
+          operation: client => session => {
```

Comment:
Is there any reason we couldn't have a single function with two parameters?  We're not currying anywhere so I don't think this is necessary

Reply:
The function returned from the first call has to be correctly shaped for withSession, withSession only passes in the session as a single argument. That being said after updating this test to the before/after style the client is accessible from the outer scope so we don't need the factory style anymore